### ISSUE TITLE
Problem with LDAP module

### ISSUE THREAD

**Issue by LeDistordu** [odoo/odoo#12883]
![chrome_2016-07-22_16-53-40](https://cloud.githubusercontent.com/assets/7805145/17061047/e8922cbc-502c-11e6-939a-ae489e9ec940.jpg)
![chrome_2016-07-22_16-56-16](https://cloud.githubusercontent.com/assets/7805145/17061109/388a931c-502d-11e6-8561-560dffd7c33e.jpg)

Impacted versions:
Version 9.0e-20160613 (Enterprise Edition)

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 643, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 680, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 316, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 309, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 959, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/openerp/http.py", line 509, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/openerp/addons/web/controllers/main.py", line 911, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/addons/web/controllers/main.py", line 903, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/models.py", line 3193, in read
    result = BaseModel.read(records, fields, load=load)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/models.py", line 3228, in read
    self._read_from_database(stored, inherited)
  File "/usr/lib/python2.7/dist-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/openerp/models.py", line 3419, in _read_from_database
    res2 = self._columns[f].get(cr, self._model, ids, f, user, context=context, values=result)
  File "/usr/lib/python2.7/dist-packages/openerp/osv/fields.py", line 1495, in get
    result = self._fnct(obj, cr, uid, ids, name, self._arg, context)
  File "/usr/lib/python2.7/dist-packages/openerp/addons/base/module/module.py", line 170, in _get_desc
    output = publish_string(source=module.description or '', settings_overrides=overrides, writer=MyWriter())
  File "/usr/lib/python2.7/dist-packages/docutils/core.py", line 414, in publish_string
    enable_exit_status=enable_exit_status)
  File "/usr/lib/python2.7/dist-packages/docutils/core.py", line 662, in publish_programmatically
    output = pub.publish(enable_exit_status=enable_exit_status)
  File "/usr/lib/python2.7/dist-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/usr/lib/python2.7/dist-packages/docutils/readers/__init__.py", line 72, in read
    self.parse()
  File "/usr/lib/python2.7/dist-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/__init__.py", line 172, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 171, in run
    input_source=document['source'])
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2727, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 328, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 396, in new_subsection
    node=section_node, match_titles=True)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 283, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2727, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 328, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 396, in new_subsection
    node=section_node, match_titles=True)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 283, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2676, in indent
    definitionlistitem, blank_finish = self.definition_list_item(context)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2797, in definition_list_item
    self.nested_parse(indented, input_offset=line_offset, node=definition)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 283, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2372, in line
    line=self.state_machine.abs_line_number())
  File "/usr/lib/python2.7/dist-packages/docutils/utils/__init__.py", line 235, in severe
    return self.system_message(self.SEVERE_LEVEL, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/docutils/utils/__init__.py", line 193, in system_message
    raise SystemMessage(msg, level)
SystemMessage: <string>:21: (SEVERE/4) Unexpected section title or transition.

---------------------
```

**Comment by Yenthe666** [comment]
Hi @LeDistordu,

I've just tested this on the latest Odoo 9.0 (ENT) and had no issues installing this module.
Please try upgrading your Odoo to the latest codebase or try reproducing this bug on the latest V9 ENT.
If you still encounter this issue then you can reply here and I'll re-open the issue.

Regards,
Yenthe

**Comment by LeDistordu** [comment]
I just update to 9.0e-20160725 version, I think it was a problem during installation but I can not seem to uninstall it to install new

![chrome_2016-07-25_15-39-30](https://cloud.githubusercontent.com/assets/7805145/17103331/24f36ca4-527e-11e6-88ee-eda67f83be63.jpg)
![chrome_2016-07-25_15-39-38](https://cloud.githubusercontent.com/assets/7805145/17103338/2a46e80c-527e-11e6-9338-e7da847ca5f2.jpg)

**Comment by LeDistordu** [comment]
This problem is driving me crazy, the module generates an error, and I can not remove it for testing a re-installation. How can I get out ?

**Comment by LeDistordu** [comment]
Same problem here  https://github.com/odoo/odoo/issues/11136

**Comment by LeDistordu** [comment]
Anyone can help me ?

**Comment by Yenthe666** [comment]
Hi @LeDistordu,

If I'm not mistaking you could just click on uninstall to cancel the module?

**Comment by LeDistordu** [comment]
I can't :(

List view and delete :
https://cloud.githubusercontent.com/assets/7805145/17103331/24f36ca4-527e-11e6-88ee-eda67f83be63.jpg

If i click on module i have : 
https://cloud.githubusercontent.com/assets/7805145/17061109/388a931c-502d-11e6-8561-560dffd7c33e.jpg

**Comment by Yotouille** [comment]
Hi there,

Is there a way to force uninstalling a module, from command line or anything ?

Thanks,
Y.

**Comment by Yenthe666** [comment]
Hi @LeDistordu,

The best way is to probably do a -u all on the whole database. This should get you going.
Be sure to first update the whole Odoo code and to take a database back-up, just in case if you run in to any problems, so you can revert.

Regards,
Yenthe

**Comment by LeDistordu** [comment]
I already tried this, it has not solved anything :/

**Comment by Yotouille** [comment]
Hi again,

I have been able to track down this issue, it's happening if you use odoo in **french**. Doing a fresh install and choosing French langage I've been able te reproduce the exact same error. In http logs I have seen something not shown in the Traceback on webUI:

```
2016-10-04 13:58:43,264 1 INFO mydb werkzeug: 172.17.0.1 - - [04/Oct/2016 13:58:43] "POST /web/dataset/call_kw/ir.module.module/fields_view_get HTTP/1.1" 200 -
<string>:2: (WARNING/2) Title underline too short.

Ajoute le support pour l'authentification par serveur LDAP.
==========================================
<string>:15: (WARNING/2) Title underline too short.

Consid\xe9rations de s\xe9curit\xe9:
------------------------
<string>:15: (WARNING/2) Title underline too short.

Consid\xe9rations de s\xe9curit\xe9:
------------------------
<string>:21: (SEVERE/4) Unexpected section title or transition.

---------------------
2016-10-04 13:58:43,437 1 ERROR mydb openerp.http: Exception during JSON request handling.
```
This is happening in the logs right before the same Traceback pasted by OP.
The accents seems to be lost in translation, as the titling mechanism in the module description.
`Consid\xe9rations de s\xe9curit\xe9` > `Considérations de sécurité`

After that, I've tried again to reinstall the whole things (pgsql+odoo) choosing US as main language, and no error shown up, indeed.

Can we get rid of this error by deleting `README.rst` file in this module ?

Thanks,
Y.

**Comment by Yotouille** [comment]
Ok, I've now pinned down the exact file doing all this mess.
`/usr/lib/python2.7/dist-packages/openerp/addons/base/i18n/fr.po`
On the line 5213, I had to rewrite the entire french description for `auth_ldap`, it was all packed on the same line with indentations errors here and there.

I've deleted the offending line 5213, and paste the new formatted description:
```
msgstr ""
"Ajoute le support pour l'authentification via serveur LDAP.\n"
"===========================================================\n"
"Ce module permet aux utilisateurs de se connecter avec leur nom d'utilisateur\n"
"et mot de passe LDAP, et il crée automatiquement les utilisateurs Odoo à la volée.\n"
"\n"
"**Note:** Ce module ne fonctionne que sur les serveurs possédant le module Python 'ldap'.\n"
"\n"
"Configuration:\n"
"--------------\n"
"Après l'installation de ce module, configurer les paramètres LDAP dans l'onglet\n"
"Configuration des détails Société. Chaque sociétés peut avoir son serveur LDAP,\n"
"tant que les noms d'utilisateurs restent uniques (les noms d'utilisateur doivent\n"
"être uniques dans Odoo, même pour plusieurs sociétés).\n"
"\n"
"La liaison LDAP anonyme est prise en charge (sur les serveurs LDAP qui le permettent),\n"
"en gardant simplement l'utilisateur LDAP avec un mot de passe vide dans la configuration\n"
"LDAP. Cela ne permet pas l'authentification anonyme pour les utilisateurs, c'est seulement\n"
"pour le compte maître LDAP utilisé pour vérifier si un utilisateur existe avant d'essayer\n"
"de l'authentifier.\n"
"\n"
"Sécuriser la connexion avec STARTTLS est disponible avec les serveurs LDAP le supportant,\n"
"pour ceci, activer l'option TLS dans la configuration LDAP.\n"
"\n"
"Pour plus d'options de configuration des paramètres LDAP, se reporter à la manpage de\n"
"ldap.conf : `ldap.conf (5)`.\n"
"\n"
"Considérations de sécurité:\n"
"---------------------------\n"
"Les mots de passe LDAP des utilisateurs ne sont jamais stockés dans la base de données\n"
"Odoo, le serveur LDAP est interrogé chaque fois qu'un utilisateur doit être authentifié.\n"
"Aucune reproduction du mot de passe se produit, et les mots de passe sont gérés en un seul\n"
"point.\n"
"\n"
"Odoo ne gère pas les changements de mot de passe dans LDAP, ainsi tout changement de mot\n"
"de passe s'introduit par d'autres moyens dans le répertoire LDAP directement (pour les\n"
"utilisateurs LDAP).\n"
"\n"
"Il est également possible d'avoir les utilisateurs Odoo locaux dans la base de données ainsi\n"
"que des utilisateurs authentifiés LDAP (le compte Administrateur est un exemple évident).\n"
"\n"
"Voici comment cela fonctionne:\n"
"------------------------------\n"
"    * Le système tente d'abord d'authentifier les utilisateurs sur la base de données Odoo\n"
"      locale ;\n"
"    * Si cette authentification échoue (par exemple parce que l'utilisateur n'a pas de mot\n"
"      de passe local), le système tente alors une authentification LDAP ;\n"
"\n"
"Comme les utilisateurs LDAP ont des mots de passe vierges par défaut dans la base de données\n"
"Odoo locale (ce qui signifie pas d'accès), la première étape est toujours un échec et le serveur\n"
"LDAP est interrogé pour procéder à l'authentification.\n"
"\n"
"L'activation de STARTTLS veille à ce que la requête d'authentification au serveur LDAP\n"
"soit cryptée.\n"
"\n"
"Modèle de l'utilisateur:\n"
"------------------------\n"
"Dans la configuration de LDAP sur le formulaire Société, il est possible de sélectionner\n"
"un *modèle utilisateur*. S'il est défini, cet utilisateur sera utilisé comme modèle pour créer\n"
"les utilisateurs locaux chaque fois que quelqu'un s'authentifie pour la première fois via une\n"
"authentification LDAP. Cela permet de pré-régler les groupes par défaut et les menus des\n"
"utilisateurs pour la première fois.\n"
"\n"
"**Attention:** si un mot de passe est donné pour le modèle d'utilisateur, ce mot de passe sera\n"
"    attribué comme mot de passe local à chaque nouvel utilisateur LDAP, mettant en place un\n"
"    *mot de passe maître* pour ces utilisateurs (jusqu'à la modification manuelle). Ceci n'est\n"
"    généralement pas recherché. On configure facilement le modèle utilisateur en se connectant\n"
"    une fois avec un utilisateur LDAP valide : laisser Odoo créer un utilisateur local vide\n"
"    avec le même login (et un mot de passe vide), puis renommer ce nouvel utilisateur avec un\n"
"    nom d'utilisateur qui ne existe pas dans LDAP, puis configurer ses groupes\n\n"
```
And now, the module does not blow up when we click on it.

Do you think this would be fixed in next version ?

Thanks,
Y.

ps: I did not success in formatting the last bloc, same problem for US version. Bold text does not finish at the right position, and indentation is not understood by parser.

**Comment by Yotouille** [comment]
And by the way, this is not the only broken french description.

Same problem here: `base.module_report_webkit`

To reproduce the same error: swith language in french, search for "webkit" in module, and click on _"Moteur de rapports du Webkit"_ module.

**Comment by Yotouille** [comment]
Another detail, it is a "blocking" problem and it is not, for the module is working perfectly but french people using odoo in their language will think the module is completely broken, not viewing the description explaining where to configure ldap support.

> Après l'installation de ce module, configurer les paramètres LDAP dans l'onglet Configuration des détails Société.

**Comment by Yotouille** [comment]
I have open a new issue about this problem finally not link to `auth_ldap`, maybe this one can be closed now.

new issue > https://github.com/odoo/odoo/issues/13695

**Comment by Yenthe666** [comment]
Hi @Yotouille,

Thank you for the further digging around and pin pointing it. 👍 
@mart-e this is translation and language related, so this is really something for you! 💎

**Comment by mart-e** [comment]
Fixed at 3f192abafbe4ffbccaba07a5f050419479d7fb7a
Thanks
